server/src/test/java/org/elasticsearch/common/settings/SettingsModuleTests.java

```diff
@@ -47,9 +51,17 @@ public void testValidate() {
         }
 
         {
-            Settings settings = Settings.builder().put("cluster.routing.allocation.balance.shard", "[2.0]")
+
+            Settings settings = Settings.builder()
+                .put("cluster.routing.allocation.balance.shard", "[2.0]")
                 .put("some.foo.bar", 1).build();
-            assertThatThrownBy(() -> new SettingsModule(settings))
+
+            // Can't directly set index.version.created since SettingsModule ctor expects only user input and rejects "index" prefixed settings.
+            // However, we have to use CURRENT version to enforce "not found setting leads to an error" behavior.
+            Settings versiondSettings = spy(settings);
```

Comment:
This doesn't feel quite right, as the SettingsModule is checking node settings not index settings?

Reply:
It checks index settings [here](https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java#L82-L85). 

If I use a regular `Settings` with a properly set version property instead of `Spy`, I get an exception 
> Index settings found. These have been unsupported since CrateDB 2.0 and should've been migrated.

Reply:
Thanks, I reverted changes in `SettingsModuleTests` and added a test.